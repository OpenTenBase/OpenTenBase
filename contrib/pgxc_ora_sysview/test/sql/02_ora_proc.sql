-- connecting superuser(default)
\connect - opentenbase
SET search_path = sys;
SELECT owner,object_type,count(*) from dba_procedures group by 1,2 order by 1;

-- connecting db using jynnuser(objects owner)
\connect - jynnuser

SET search_path = sys;
SELECT owner,object_type,count(*) from all_procedures group by 1,2 order by 1;

SELECT object_type,count(*) from user_procedures group by 1 order by 1;

-- connecting db using zjaauser(Only able to access some objects under the schema jynnn)
\connect - zjaauser
SET search_path = sys;
SELECT owner,object_type,count(*) from all_procedures group by 1,2 order by 1;

SELECT object_type,count(*) from user_procedures group by 1 order by 1;