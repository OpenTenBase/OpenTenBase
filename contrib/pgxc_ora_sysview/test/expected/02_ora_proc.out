joyann=> \connect - opentenbase
You are now connected to database "joyann" as user "opentenbase".
joyann=# SET search_path = sys;
SET
joyann=# SELECT owner,object_type,count(*) from dba_procedures group by 1,2 order by 1;
    owner    | object_type | count 
-------------+-------------+-------
 jynnuser    | FUNCTION    |     2
 opentenbase | FUNCTION    |  3519
(2 rows)

joyann=# \connect - jynnuser
You are now connected to database "joyann" as user "jynnuser".
joyann=> 
joyann=> SET search_path = sys;
SET
joyann=> SELECT owner,object_type,count(*) from all_procedures group by 1,2 order by 1;
    owner    | object_type | count 
-------------+-------------+-------
 jynnuser    | FUNCTION    |     3
 opentenbase | FUNCTION    |  3497
(2 rows)

joyann=> 
joyann=> SELECT object_type,count(*) from user_procedures group by 1 order by 1;
 object_type | count 
-------------+-------
 FUNCTION    |     3
(1 row)


joyann=> \connect - zjaauser
You are now connected to database "joyann" as user "zjaauser".
joyann=> SET search_path = sys;
SET
joyann=> SELECT owner,object_type,count(*) from all_procedures group by 1,2 order by 1;
    owner    | object_type | count 
-------------+-------------+-------
 jynnuser    | FUNCTION    |     3
 opentenbase | FUNCTION    |  3497
(2 rows)

joyann=> 
joyann=> SELECT object_type,count(*) from user_procedures group by 1 order by 1;
 object_type | count 
-------------+-------
(0 rows)
